src/test/blockfilter_index_tests.cpp

```diff
@@ -175,7 +175,7 @@ BOOST_FIXTURE_TEST_CASE(blockfilter_index_initial_sync, TestChain100Setup)
     uint256 chainA_last_header = last_header;
     for (size_t i = 0; i < 2; i++) {
         const auto& block = chainA[i];
-        BOOST_REQUIRE(ProcessNewBlock(Params(), block, true, nullptr));
```

Comment:
In commit "Make ProcessNewBlock return a future instead of an immediate bool" (79e6e446dd0c8a3f53759a9792388f28e1053bfb)

In the commit message about these tests you wrote "Note that this makes some of the unit tests less restrictive." But I don't think there's a good justification for doing this. If you just declared `struct ProcessNewBlockResult { bool error = false; bool already_have = false; };` and returned `std::future<ProcessNewBlockResult>`, the tests could behave the same and the code would be more readable and less fragile, without the need for people to guess from context whether the bool indicates success or newness.

Reply:
Think I agree with this suggestion; dropping the `fNewBlock` result at the same time as turning the return into a `future` seems unnecessarily complicated.